pkg/resources/server_test.go

```diff
@@ -0,0 +1,327 @@
+				event := fsnotify.Event{Name: "fake", Op: fsnotify.Create}
+				rs.socketWatcher.Events <- event
+			}()
+			err := rs.Watch()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("unable to restart server failed"))
+		})
+		It("Watch socket and getting failed error from watcher", func() {
+			rs := resourceServer{}
+			go func() {
+				// Wait for watch to create the socket watcher
+				time.Sleep(5 * time.Millisecond)
+				err := errors.New("failed")
+				rs.socketWatcher.Errors <- err
+			}()
+			err := rs.Watch()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("inotify: failed"))
+		})
+		It("Watch socket then stop watcher", func() {
+			rs := resourceServer{stopWatcher: make(chan bool)}
+			go func() {
+				// Wait for watch to create the socket watcher
+				time.Sleep(5 * time.Millisecond)
+				rs.stopWatcher <- true
+			}()
+			err := rs.Watch()
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+	Context("ListAndWatch", func() {
+		It("GetInfo of plugin", func() {
+			rs := resourceServer{
+				stop:   make(chan interface{}),
+				health: make(chan *pluginapi.Device),
+			}
+			// Dummy sender
+			go func() {
+				rs.health <- &pluginapi.Device{}
+				// Make sure that health call before the stop
+				time.Sleep(1 * time.Millisecond)
+				rs.stop <- "stop"
+			}()
+			err := rs.ListAndWatch(nil, &devPluginListAndWatchServerMock{})
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+	Context("Allocate", func() {
+		It("Allocate resource", func() {
+			rs := resourceServer{resourceName: "fake", endPoint: "fake.sock"}
+			req := &pluginapi.AllocateRequest{
+				ContainerRequests: []*pluginapi.ContainerAllocateRequest{nil, nil},
+			}
+			res, err := rs.Allocate(context.TODO(), req)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(res.ContainerResponses)).To(Equal(2))
+		})
+	})
+	Context("GetInfo", func() {
+		It("GetInfo of plugin", func() {
+			rs := resourceServer{resourceName: "fake", endPoint: "fake.sock"}
+			resp, err := rs.GetInfo(context.TODO(), nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(resp.Type).To(Equal(registerapi.DevicePlugin))
+			Expect(resp.Name).To(Equal("fake"))
+			Expect(resp.Endpoint).To(Equal("/var/lib/kubelet/plugins_registry/fake.sock"))
+		})
+	})
+	Context("NotifyRegistrationStatus", func() {
+		It("NotifyRegistrationStatus with plugin registered", func() {
+			regstat := &registerapi.RegistrationStatus{PluginRegistered: true}
+			rs := resourceServer{endPoint: "fake.sock"}
+			_, err := rs.NotifyRegistrationStatus(context.TODO(), regstat)
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("NotifyRegistrationStatus with plugin unregistered", func() {
+			regstat := &registerapi.RegistrationStatus{PluginRegistered: false}
+			rsc := &mocks.ResourceServerConnector{}
+			rsc.On("Stop").Return()
+			rs := resourceServer{endPoint: "fake.sock", rsConnector: rsc}
+			_, err := rs.NotifyRegistrationStatus(context.TODO(), regstat)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
can you check if server.Stop() was called ?

Reply:
added AssertExpectations